assimilation_code/modules/utilities/parse_args_mod.f90

```diff
@@ -554,7 +587,407 @@ subroutine get_next_arg(inline, startoff, argword, endoff)
 
 end subroutine get_next_arg
 
-!------------------------------------------------------------------------------
+
+!--------------------------------------------------------------
+! Retrieve a string column using a cached handle.
+subroutine csv_get_field_char(cf, varname, varvals, context)
+
+type(csv_file_type), intent(in)  :: cf
+character(len=*),    intent(in)  :: varname
+character(len=*),    intent(out) :: varvals(:)
+character(len=*),    intent(in), optional :: context
+
+character(len=*), parameter :: routine = 'csv_get_field_char'
+
+integer                       :: fidx
+integer                       :: iunit, io, iobs, nfields
+character(len=MAX_FIELDS_LEN) :: line
+character(len=512)            :: entries(MAX_NUM_FIELDS)
+
+if (.not. cf%is_open) then
+   string1 = 'CSV file handle has not been initialized.'
+   call error_handler(E_ERR, routine, string1, context)
+   return
+endif
+
+! Locate field index in cached header
+fidx = csv_find_field(cf, varname)
+if (fidx < 1 .or. fidx > cf%ncols) then
+   string1 = 'Requested field "'//trim(varname)//'" not found in header for file "'// &
+             trim(cf%filename)//'"'
+   call error_handler(E_ERR, routine, string1, context)
+endif
+
+! Open file and skip header
+iunit = open_file(cf%filename, action='read', form='formatted')
```

Comment:
if you leave the file open, you can call a new "skip_header()" routine which does a "rewind" and reads the first line.  you already know the delimiter and you've already parsed the header so you can just discard the line.